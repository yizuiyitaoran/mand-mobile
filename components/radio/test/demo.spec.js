import Demo0 from './cases/demo0'
import Demo1 from './cases/demo1'
import Demo2 from './cases/demo2'
import Demo3 from './cases/demo3'
import Demo4 from './cases/demo4'
import Demo5 from './cases/demo5'
import {renderToString} from '@vue/server-test-utils'

describe('Radio', () => {
  test(`Basic`, () => {
    expect(renderToString(Demo0)).toMatchSnapshot()
  })
  test(`With Field`, () => {
    expect(renderToString(Demo1)).toMatchSnapshot()
  })
  test(`List`, () => {
    expect(renderToString(Demo2)).toMatchSnapshot()
  })
  test(`Radio group`, () => {
    expect(renderToString(Demo3)).toMatchSnapshot()
  })
  test(`Radio box`, () => {
    expect(renderToString(Demo4)).toMatchSnapshot()
  })
  test(`Radio box group`, () => {
    expect(renderToString(Demo5)).toMatchSnapshot()
  })
})
