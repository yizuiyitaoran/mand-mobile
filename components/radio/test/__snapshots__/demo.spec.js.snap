// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Basic 1`] = `
<div class="md-example-child md-example-child-radio md-example-child-radio-0"><label class="md-radio is-checked">
    <div class="md-radio-icon"><i class="md-icon icon-font md-icon-checked checked md" style="color:;"></i></div>
    <div class="md-radio-label">单选项1</div>
  </label> <label class="md-radio">
    <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check md" style="color:;"></i></div>
    <div class="md-radio-label">单选项2</div>
  </label></div>
`;

exports[`Radio List 1`] = `
<div class="md-example-child md-example-child-radio md-example-child-radio-2">
  <fieldset class="md-field radio-field">
    <header class="md-field-header">
      <div class="md-field-heading">
        <legend class="md-field-title">简单选择列表</legend>
        <!---->
      </div>
      <div class="md-field-action"></div>
    </header>
    <div class="md-field-content">
      <div class="md-radio-list">
        <div class="md-cell-item md-radio-item">
          <div class="md-cell-item-body multilines">
            <div class="md-cell-item-left"><label class="md-radio">
                <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check lg" style="color:;"></i></div>
                <!---->
              </label></div>
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">交通银行(尾号3089)</p>
              <p class="md-cell-item-brief">选项摘要描述</p>
            </div>
            <!---->
          </div>
          <!---->
        </div>
        <div class="md-cell-item md-radio-item">
          <div class="md-cell-item-body multilines">
            <div class="md-cell-item-left"><label class="md-radio">
                <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check lg" style="color:;"></i></div>
                <!---->
              </label></div>
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">招商银行(尾号2342)</p>
              <p class="md-cell-item-brief">选项摘要描述</p>
            </div>
            <!---->
          </div>
          <!---->
        </div>
        <div class="md-cell-item md-radio-item is-disabled no-border">
          <div class="md-cell-item-body multilines">
            <div class="md-cell-item-left"><label class="md-radio is-disabled">
                <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check-disabled check-disabled lg" style="color:;"></i></div>
                <!---->
              </label></div>
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">建设银行(尾号4321)</p>
              <p class="md-cell-item-brief">选项摘要描述</p>
            </div>
            <!---->
          </div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
    <!---->
  </fieldset>
  <fieldset class="md-field radio-field">
    <header class="md-field-header">
      <div class="md-field-heading">
        <legend class="md-field-title">输入项</legend>
        <!---->
      </div>
      <div class="md-field-action"></div>
    </header>
    <div class="md-field-content">
      <div class="md-radio-list">
        <div class="md-cell-item md-radio-item">
          <div class="md-cell-item-body">
            <!---->
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">我有其他保险</p>
              <!---->
            </div>
            <div class="md-cell-item-right"><label class="md-radio">
                <div class="md-radio-icon">
                  <!---->
                </div>
                <!---->
              </label>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="md-cell-item md-radio-item">
          <div class="md-cell-item-body">
            <!---->
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">每单都扣，我担心扣太多</p>
              <!---->
            </div>
            <div class="md-cell-item-right"><label class="md-radio">
                <div class="md-radio-icon">
                  <!---->
                </div>
                <!---->
              </label>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="md-cell-item md-radio-item">
          <div class="md-cell-item-body">
            <!---->
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">我身体好，不需要重疾险</p>
              <!---->
            </div>
            <div class="md-cell-item-right"><label class="md-radio">
                <div class="md-radio-icon">
                  <!---->
                </div>
                <!---->
              </label>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="md-cell-item md-radio-item">
          <div class="md-cell-item-body">
            <!---->
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">接单少，加入这个没什么用</p>
              <!---->
            </div>
            <div class="md-cell-item-right"><label class="md-radio">
                <div class="md-radio-icon">
                  <!---->
                </div>
                <!---->
              </label>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="md-cell-item md-radio-item no-border">
          <div class="md-cell-item-body">
            <!---->
            <div class="md-cell-item-content">
              <p class="md-cell-item-title">我不了解这是什么计划</p>
              <!---->
            </div>
            <div class="md-cell-item-right"><label class="md-radio">
                <div class="md-radio-icon">
                  <!---->
                </div>
                <!---->
              </label>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="md-field-item md-input-item md-radio-item is-solid is-browser left normal">
          <div class="md-field-item-content"><label class="md-field-item-title">其他</label>
            <!---->
            <div class="md-field-item-control"> <input name="" placeholder="请输入原因" maxlength="" autocomplete="off" type="text" value="" class="md-input-item-input"> </div>
            <!---->
          </div>
          <!---->
        </div>
      </div>
    </div>
    <!---->
  </fieldset>
</div>
`;

exports[`Radio Radio box 1`] = `
<div class="md-example-child md-example-child-check md-example-child-check-4">
  <div class="md-check-base-box md-radio-box is-disabled">日缴
    <!---->
  </div>
  <div class="md-check-base-box md-radio-box">月付
    <!---->
  </div>
  <div class="md-check-base-box md-radio-box">季度费
    <!---->
  </div>
</div>
`;

exports[`Radio Radio box group 1`] = `
<div class="md-example-child md-example-child-check md-example-child-check-5">
  <div class="md-radio-group">
    <div class="md-check-base-box md-radio-box is-checked">自己 <div class="md-tag">
        <div class="default size-tiny shape-quarter type-fill font-weight-normal">
          <div class="quarter-content"><i class="md-icon icon-font md-icon-right right md" style="color:;"></i></div>
          <div class="quarter-bg"></div>
        </div>
      </div>
    </div>
    <div class="md-check-base-box md-radio-box">配偶
      <!---->
    </div>
    <div class="md-check-base-box md-radio-box">父母
      <!---->
    </div>
    <div class="md-check-base-box md-radio-box is-disabled">子女
      <!---->
    </div>
  </div>
</div>
`;

exports[`Radio Radio group 1`] = `
<div class="md-example-child md-example-child-radio md-example-child-radio-3">
  <div class="md-radio-group"><label class="md-radio">
      <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check md" style="color:;"></i></div>
      <div class="md-radio-label">西瓜</div>
    </label> <label class="md-radio is-checked">
      <div class="md-radio-icon"><i class="md-icon icon-font md-icon-checked checked md" style="color:;"></i></div>
      <div class="md-radio-label">苹果</div>
    </label> <label class="md-radio">
      <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check md" style="color:;"></i></div>
      <div class="md-radio-label">香蕉</div>
    </label> <label class="md-radio">
      <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check md" style="color:;"></i></div>
      <div class="md-radio-label">橙子</div>
    </label> <label class="md-radio is-disabled">
      <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check-disabled check-disabled md" style="color:;"></i></div>
      <div class="md-radio-label">西红柿</div>
    </label></div>
</div>
`;

exports[`Radio With Field 1`] = `
<div class="md-example-child md-example-child-radio md-example-child-radio-1">
  <fieldset class="md-field">
    <!---->
    <div class="md-field-content">
      <div class="md-field-item is-solid is-browser">
        <div class="md-field-item-content"><label class="md-field-item-title">婚姻状况</label>
          <!---->
          <div class="md-field-item-control"><label class="md-radio is-checked is-inline">
              <div class="md-radio-icon"><i class="md-icon icon-font md-icon-checked checked md" style="color:;"></i></div>
              <div class="md-radio-label">已婚</div>
            </label> <label class="md-radio is-inline">
              <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check md" style="color:;"></i></div>
              <div class="md-radio-label">未婚</div>
            </label> <label class="md-radio is-inline">
              <div class="md-radio-icon"><i class="md-icon icon-font md-icon-check check md" style="color:;"></i></div>
              <div class="md-radio-label">保密</div>
            </label></div>
          <!---->
        </div>
        <!---->
      </div>
    </div>
    <!---->
  </fieldset>
</div>
`;
